src/awkward/forms/form.py

```diff
@@ -437,29 +437,59 @@ def columns(self, list_indicator=None, column_prefix=()):
     def select_columns(
         self, specifier, expand_braces=True, *, prune_unions_and_records: bool = True
     ):
+        """
+        select_columns returns a new Form with only columns and sub-columns selected.
+        Returns an empty Form if no columns matched the specifier(s).
+
+        `specifier` can be a `str | Iterator[str | list[str] | tuple[str]]`.
+        Strings may include shell-globbing-style wildcards "*" and "?".
+        If `expand_braces` is `True` (the default), strings may include alternatives in braces.
+        For example, `["a.{b,c}.d"]` is equivalent to `["a.b.d", "a.c.d"]`.
+        Glob-style matching would also suit this single-character instance: `"a.[bc].d"`.
+        If specifier is a list which contains a list/tuple, that inner list will be interpreted as
+        column and subcolumn specifiers. They *may* contain wildcards, but "." will not be
+        interpreted as a `<field>.<subfield>` pattern.
+        """
         if isinstance(specifier, str):
             specifier = {specifier}
 
         # Only take unique specifiers
         for item in specifier:
-            if not isinstance(item, str):
+            if isinstance(item, str):
+                if item == "":
+                    raise ValueError(
+                        "A column-selection specifier cannot be an empty string"
+                    )
+            elif isinstance(item, (tuple, list)):
```

Comment:
```suggestion
            elif isinstance(item, Iterable):
```

and

```python
from collections.abc import Iterable
```

The danger here is that `str` is also `Iterable`, but it's okay if you check for `Iterable` after checking for `str`.

This way, if someone has some other type, like a `set`, it will still work. It's not great that `dict` is also interpreted as an iterable (of its keys only), but at least that's Python convention and expected.